auth_saml/models/auth_saml.py

```diff
@@ -1,78 +1,331 @@
+# Copyright (C) 2020 Glodo UK <https://www.glodo.uk/>
 # Copyright (C) 2010-2016 XCG Consulting <http://odoo.consulting>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 import json
+import logging
+import urllib.parse
 
-import lasso
+import saml2
+from saml2.client import Saml2Client
+from saml2.config import Config as Saml2Config
 
 from odoo import api, fields, models
 
+_logger = logging.getLogger(__name__)
+
 
 class AuthSamlProvider(models.Model):
     """Configuration values of a SAML2 provider"""
-    _name = 'auth.saml.provider'
-    _description = 'SAML2 provider'
-    _order = 'sequence, name'
 
-    # Name of the OAuth2 entity, authentic, xcg...
-    name = fields.Char('Provider name', required=True, index=True)
+    _name = "auth.saml.provider"
+    _description = "SAML2 provider"
+    _order = "sequence, name"
+
+    name = fields.Char("Provider name", required=True, index=True)
+
+    entity_id = fields.Char(
+        "Entity ID",
+        help=("EntityID passed to IDP, used to identify the Odoo"),
+        required=True,
+        default="odoo",
+    )
+
     idp_metadata = fields.Text(
-        string='IDP Configuration',
-        help="Configuration for this Identity Provider",
+        string="Identity Provider Metadata",
+        help=(
+            "Configuration for this Identity Provider. Supplied by the"
+            " provider, in XML format."
+        ),
     )
-    sp_metadata = fields.Text(
-        string='SP Configuration',
-        help="Configuration for the Service Provider (this Odoo instance)",
+
+    sp_baseurl = fields.Text(
+        string="Override Base Url",
+        help="""Base Url sent to Odoo with this, rather than automatically
+        detecting from request or system parameter web.base.url
+        """,
     )
-    sp_pkey = fields.Text(
-        string='SP private key',
-        help="Private key for the Service Provider (this Odoo instance)",
+
+    sp_pem = fields.Binary(
+        string="Odoo Certificate",
+        help="Combined certificate and private key used by Odoo to sign",
```

Comment:
Can you state "pem" in the help text to help explaining what kind of file is expected?